src/AspNetCoreModuleV2/AspNetCore/src/applicationinfo.cpp

```diff
@@ -218,26 +219,40 @@ APPLICATION_INFO::EnsureApplicationCreated(
         //
         if (!m_fAppOfflineFound)
         {
-
+            m_appStatus = APPLICATION_STATUS::FAIL;
```

Comment:
Why don't you set it in `if (FAILED(hr)):` ?

Reply:
Also does it go from STARTING to FAIL to RUNNING?

Reply:
the initial state of applicationinfo (during creation) is FAILED, we then do not need to set FAILED at finished which is not guarded by lock. The status goes from FAILED to STARTING or RUNNING.

Reply:
It's not what I see in code. 

OnExecuteRequestHandler calls GetOrCreateApplicationInfo that sets the status to STARTING then EnsureApplicationCreated gets called setting status to FAILED and then back to RUNNING. 

Did I miss something?

Reply:
the first one is the status of applicationinfo (status is member element of applicationinfo) and the second one is status of application. I just reused the enum type.

Reply:
my fault. I had thought I could use one enum to replace the two BOOLs in previous pr.  Revert this enum change.